src/graphics/Window.zig

```diff
@@ -260,7 +268,10 @@ pub const Gamepad = struct {
 	}
 
 	pub fn init() void {
-		downloadControllerMappings();
+		controllerConnectedPreviously = isControllerConnected();
+		if (controllerConnectedPreviously) {
+			downloadControllerMappings();
+		}
```

Comment:
Why even run this in the init, if you check it on every update anyways? This should allow you to fix the other problem more smoothly as well.